apps/api-extractor/src/collector/MessageRouter.ts

```diff
@@ -222,6 +242,57 @@ export class MessageRouter {
     }
   }
 
+  /**
+   * Recursively collects the primitive members (numbers, strings, arrays, etc) into an object that
+   * is JSON serializable.  This is used by the "--diagnostics" feature to dump the state of configuration objects.
+   *
+   * @returns a JSON serializable object (possibly including `null` values)
+   *          or `undefined` if the input cannot be represented as JSON
+   */
+  // tslint:disable-next-line:no-any
+  public static buildJsonDumpObject(input: any): any | undefined {
```

Comment:
It'd be useful to have some unit tests for this.

Reply:
I considered promoting this to a real utility in **node-core-library**, in which case I definitely would have added tests.

However, the concept of `buildJsonDumpObject()` seems lazy and half-baked.  A  "correct" implementation would use `JSON.serialize()` for actual data objects, or for other objects we should implement a proper printer/serializer (that knows all the members and how to format them).  In my example output, you can see that dumped options aren't very readable (e.g. `"target": 2`).

I took the lazy approach because I didn't want to spend too much time on this `--diagnostics` feature initially.  I know that we need some way to investigate API Extractor issues without resorting to a debugger. But this PR is merely a guess about what logging is actually useful.